pkg/query-service/app/queryBuilder/query_builder.go

```diff
@@ -152,11 +153,28 @@ func (qb *QueryBuilder) PrepareQueries(params *v3.QueryRangeParamsV3, args ...in
 					}
 					queries[queryName] = queryString
 				case v3.DataSourceLogs:
-					queryString, err := qb.options.BuildLogQuery(params.Start, params.End, compositeQuery.QueryType, compositeQuery.PanelType, query)
-					if err != nil {
-						return nil, err
+					// if panel type is timeseries/ table, handle limit differently using two queries
+					if (compositeQuery.PanelType == v3.PanelTypeGraph || compositeQuery.PanelType == v3.PanelTypeTable) && query.Limit > 0 && len(query.GroupBy) > 0 {
```

Comment:
I don't think we need to handle the table view separately like we do for the time series since there is going to be only one for each unique label set.

Reply:
Ahh with the new changes we have removed timestamp from the result so we don't need it in table view. Is that what you are trying to say ?

cc @makeavish

Reply:
Yes